kuma/core/urlresolvers.py

```diff
@@ -134,6 +134,26 @@ def get_language(self):
 
         return settings.LANGUAGE_CODE
 
+    def get_user_chosen_language(self):
+        """
+        If the user is not logged in and have a cookie set for language, return that language
+        For logged in user, return his language pointed in his profile locale.
+        Otherwise, return None
+        """
+        user = getattr(self.request, 'user', None)
+
+        if user:
+            if user.is_authenticated():
+                # The user is authenticated. So return if he have any locale set up
+                return user.locale
+            elif user.is_anonymous():
+                # The user is not authenticated. So find if he have a cookie for language preference
+                language = self.request.session.get(settings.LANGUAGE_COOKIE_NAME)
```

Comment:
Anonymous users should not have sessions. We want to store this feature in a cookie,